src/main/scala/mesosphere/marathon/core/task/update/impl/TaskStatusUpdateProcessorImpl.scala

```diff
@@ -72,18 +82,28 @@ class TaskStatusUpdateProcessorImpl @Inject() (
 object TaskStatusUpdateProcessorImpl {
   lazy val name = Names.named(getClass.getSimpleName)
 
+  /** Matches all states that are considered terminal for an unknown task */
+  private[impl] def terminalUnknown(status: MarathonTaskStatus): Boolean = status match {
+    case t: MarathonTaskStatus.Terminal => true
+    case MarathonTaskStatus.Unreachable => true
+    case _ => false
+  }
+
   private[this] val ignoreWhenUnknown = Set(
-    MesosProtos.TaskState.TASK_KILLED,
-    MesosProtos.TaskState.TASK_KILLING,
-    MesosProtos.TaskState.TASK_ERROR,
-    MesosProtos.TaskState.TASK_FAILED,
-    MesosProtos.TaskState.TASK_FINISHED,
-    MesosProtos.TaskState.TASK_LOST
+    MarathonTaskStatus.Killed,
+    MarathonTaskStatus.Killing,
+    MarathonTaskStatus.Error,
+    MarathonTaskStatus.Failed,
+    MarathonTaskStatus.Finished,
+    MarathonTaskStatus.Unreachable,
+    MarathonTaskStatus.Gone,
+    MarathonTaskStatus.Dropped,
+    MarathonTaskStatus.Unknown
   )
   // It doesn't make sense to kill an unknown task if it is in a terminal or killing state
   // We'd only get another update for the same task
-  private def killWhenUnknown(status: MesosProtos.TaskStatus): Boolean = {
-    !ignoreWhenUnknown.contains(status.getState)
+  private def killWhenUnknown(status: MarathonTaskStatus): Boolean = {
+    !ignoreWhenUnknown.contains(status)
```

Comment:
I didn't address these comments here so we can proceed with cherry-picking into 1.3. I agree we should clarify naming and use-case; let's do that in a separate PR.